android/core/src/main/java/com/stadiamaps/ferrostar/core/extensions/TripStateExtensions.kt

```diff
@@ -41,3 +45,26 @@ fun TripState.deviation() =
       is TripState.Complete,
       TripState.Idle -> null
     }
+
+/**
+ * Get the route annotations at the user's current location.
+ *
+ * This function will deserialize the annotation bytes provided by the trip
+ * state.
+ *
+ * @param type The json data class that annotations should be deserialized to.
+ * @return The annotation data or null if the trip is not navigating.
+ */
+inline fun <reified T> TripState.annotation(type: Class<T>): T? =
```

Comment:
Conceptually, we want a developer to be able to specify their json type for annotations that includes whatever their API returns. This on get method shows that, but we may want to change to a pre-parsed system. Where the annotation object's `type: Class<T>` and json adapter is added as a param to any RouteAdapter. The route adapter is then responsible for pre-parsing the data. Not sure how easy this would be to get out here as the typed value though?

Reply:
This is deferred to https://github.com/stadiamaps/ferrostar/issues/271